frontend/src/components/shared/GradeChip/GradeChip.tsx

```diff
@@ -7,8 +7,10 @@ import { useDialog } from '../ConfirmDialog/ConfirmDialogProvider';
 interface GradeChipProps {
   score: number | undefined;
   maxScore: number;
+  // readOnly determines whether the grade can be changed
```

Comment:
nit: perhaps "readOnly indicates if the grade can be changed at all depending on permissions; inEditMode tells the GradeChip component whether it should render a textfield or a string"